models/web/scratch/interactions_this_run/snowflake/snowplow_media_player_interactions_this_run.sql

```diff
@@ -26,8 +26,10 @@ with prep as (
     e.derived_tstamp as start_tstamp,
     contexts_com_snowplowanalytics_snowplow_media_player_1[0]:currentTime::float as player_current_time,
     coalesce(contexts_com_snowplowanalytics_snowplow_media_player_1[0]:playbackRate::varchar, 1) as playback_rate,
-    case when e.unstruct_event_com_snowplowanalytics_snowplow_media_player_event_1:type::varchar = 'ended' then 100 else contexts_com_snowplowanalytics_snowplow_media_player_1[0]:percentProgress::int end percent_progress,
-    contexts_com_snowplowanalytics_snowplow_media_player_1[0]:muted::varchar as is_muted,
+    cast(case when e.unstruct_event_com_snowplowanalytics_snowplow_media_player_event_1:type::varchar = 'ended' then '100'
+        when contexts_com_snowplowanalytics_snowplow_media_player_1[0]:percentProgress::varchar = '' THEN NULL
```

Comment:
When would we have NULL progress? Is that equivalent to no progress?

Reply:
The percent progress is a context field so it will always be present even when it is not a percent progress event, in which case it defaults to '' in case of Snowflake as part of the JSON string.